zanshinsdk/client.py

```diff
@@ -943,6 +943,19 @@ def get_organization_scan_target_scan(self, organization_id: Union[UUID, str], s
                              f"/organizations/{validate_uuid(organization_id)}/scantargets/"
                              f"{validate_uuid(scan_target_id)}/scans/{scan_id}").json()
 
+    ###################################################
+    # Organization Scan Target
+    ###################################################
+
+    def iter_organization_scan_target_groups(self, organization_id: Union[UUID, str]) -> Iterator[Dict]:
+        """
+        Iterates over the scan targets groups.
+        <https://api.zanshin.tenchisecurity.com/#operation/getOrganizationScanTargetGroups>
+        :param organization_id: the ID of the organization
+        : return: an iterator over the scan target groups
+        """
+        yield from self._request("GET", f"/organizations/{validate_uuid(organization_id)}/scantargetgroups").json()
```

Comment:
Makes me sad that this API operation is not paginated. :( But the SDK code is right.

Reply:
It's in plan to do but it will be a breakchange

Reply:
> It's in plan to do but it will be a breakchange

This is one (of many) we won't be able to avoid. Customers can have an arbitrarily large number of scan targets and scan target groups.

We can mitigate this by using sensible defaults. For example page defaulting to `1` and `pageSize`  defaulting to a very big number that would make existing customers still getting all of their scan targets if those two parameters are not provided.

Reply:
The issue will be the output because it returns an array of organization `Array<T>` and it will need to return a object with array `{data:Array<T>, cursor: string}`, so that will be a break change.
Also dynamo supports only cursor for pagination so wouldn't be possible using `page`

Reply:
Yeah. that sucks. We'll need to figure out how to handle that correctly. Maybe use different stages for a "v2" endpoint with pagination and leave the old one as-is. What a mess.